### PR TITLE
Add random_exclude_pattern option for random songs

### PR DESCRIPTION
**Pull request by spk**
Hi, this allow to exclude with a pattern some songs from add random songs action, cheers

### CODE REVIEW

**Comment by heiderich** [comment]
I very much welcome this feature. I did some quick tests and it seems to work.

File: src/settings.h [hunk 1]
```diff
@@ -178,6 +178,7 @@ struct Configuration
 
 	unsigned mpd_connection_timeout;
 	unsigned crossfade_time;
+	std::string random_exclude_pattern;
```
Comment (larson-carter) [inline src/settings.h]: This is a little bit out of place. Yes it works but can you please reformat this? Line 208 would be the proper place for it.

**Comment by larson-carter** [comment]
> Hi, this allow to exclude with a pattern some songs from add random songs action, cheers

This will be approved as soon as you fix the out of place declaration on line 181 in file src/settings.h.

File: src/settings.h (hunk 1 above)
Comment (spk) [reply src/settings.h]: > This is a little bit out of place. Yes it works but can you please reformat this? Line 208 would be the proper place for it.

Sure done in ee79bfd

Comment (larson-carter) [reply src/settings.h]: Thank you.

**Pull request merged**